chore: add beginning of migration guide for Cypress 15 (#6159)
cypress-io/cypress-documentation: docs/app/references/migration-guide.mdx

# Migration Guide

## Migrating to Cypress 15.0

This guide details the code changes needed to migrate to Cypress
version 15.
[See the full changelog for version v15.0](/app/references/changelog#15-0-0).

### Node.js 20 and 22+ support

> Cypress comes bundled with its own
> [Node.js version](https://github.com/cypress-io/cypress/blob/develop/.node-version).
> However, installing the `cypress` npm package uses the Node.js version installed
> on your system.
> 
> [See Node's release schedule](https://github.com/nodejs/Release). Node.js
> version 18 and 23 are no longer supported when installing Cypress. The supported Node.js
> versions to install Cypress are now Node.js 20 and 22+.

Reviewer: This wording has been carried over from the Cypress 14 migration guide.

1. The link to .node-version is to the current version in the `develop` branch (`20.18.1`), which is not necessarily the same version of Node.js that is going to be shipped with each Cypress 15.x release. The exact version can be shown with `cypress version --component node`, however that information isn't really necessary here.
2. Strictly speaking, the `cypress` npm package can be installed without Node.js. It's the `postinstall` script of the `cypress` npm package that needs Node.js to install the Cypress binary.

I've made a suggestion to simplify this section and make it more accurate:

```suggestion
Cypress requires [Node.js](https://nodejs.org/en) in order to install the Cypress binary and the supported versions are now Node.js 20 and 22+.
Node.js versions 18 and 23 are no longer supported.
[See Node's release schedule](https://github.com/nodejs/Release).
```

![image](https://github.com/user-attachments/assets/974a7498-bb18-48ae-b423-03631c707d0e)

> [See Node's release schedule](https://github.com/nodejs/Release). Node.js
> version 18 and 23 will no longer be supported when installing Cypress. The minimum Node.js
> version supported to install Cypress is Node.js 20 and 22+.

Reviewer: ```suggestion
[See Node's release schedule](https://github.com/nodejs/Release). Node.js
version 18 and 23 are no longer supported when installing Cypress. The supported Node.js
versions to install Cypress are now Node.js 20 and 22+.
```

You can't really have two minimums.
Node.js v24.0.0 was supposed be released today, however they are having compatibility issues with a newer V8 engine and the release is delayed. I'll drop a note if there is any new planning date available.

Author: good point. Once it releases we need to test Cypress out with it to see if anything is broken
